### PR TITLE
Support sticky session cookies for TUS uploads

### PR DESCRIPTION
**Pull request by MPvHarmelen**

### CODE REVIEW

File: src/utils/index.ts [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
 export function wait(ms: number) {
   return new Promise<void>(function (resolve) {
     setTimeout(function () {
       resolve();
     }, ms);
   });
 }
+
+/**
+ * Attempt to set a stick session cookie in the given headers object. If this
```
Comment (thombohlk) [inline src/utils/index.ts:13]: :pick: `stickY`

File: src/Dataset.ts [hunk 2]
```diff
@@ -693,21 +693,27 @@ export class JobUpload {
...
-        },
+        headers,
         mapUrl: this.urlMapper,
         chunkSize: 5 * 1024 * 1024,
         uploadSize: fileSize,
         retryDelays: [2000, 5000, 10000, 40000, 50000],
         onError: reject,
-        onProgress: function (_bytesUploaded: number, _bytesTotal: number) {},
+        onChunkComplete: function (...stats) {
+          log(formatUploadProgress(Date.now() - previousChunkCompleted, ...stats));
```
Comment (thombohlk) [inline src/Dataset.ts:714]: :thinking: For really large uploads this might flood the logs in pipelines. I think given the data you could also determine when the rounded percentage increases and only then log the progress. Do you think that is worth implementing?

File: src/utils/index.ts [hunk 3]
```diff
@@ -1,7 +1,48 @@
...
+  // the time difference is in ms, so the averageSpeed is in bytes / ms = kB / s
+  const uploadSpeed = chunkSize / msSinceLastChunk;
+  const formattedUploadSpeed =
+    uploadSpeed > 1000 ? (uploadSpeed / 1000).toFixed(1) + "M" : uploadSpeed.toFixed(1) + "k";
+  return `uploaded ${bytesUploaded
+    .toLocaleString("en")
+    .padStart(
+      formattedBytesTotal.length
+    )} of ${formattedBytesTotal} bytes (${percentage}%, ~${formattedUploadSpeed}B/s)`;
+}
```
Comment (thombohlk) [inline src/utils/index.ts:48]: :smile:

File: src/__tests__/Asset-test.ts [hunk 4]
```diff
@@ -45,6 +45,11 @@ describe("Assets", function () {
     const newAsset = await testDs.uploadAsset("./README.md", this.test?.title);
     expect(newAsset.getInfo().fileSize).to.equal((await fs.stat("./README.md")).size);
   });
+  it("Adding a large asset", async function () {
```
Comment (thombohlk) [inline src/__tests__/Asset-test.ts:48]: :wrench: Shouldn't this one also skip?

File: tsconfig.json [hunk 5]
```diff
@@ -2,7 +2,7 @@
   "compilerOptions": {
     "target": "es2020",
     "lib": ["es2020", "esnext"],
-    "module": "es2020",
+    "module": "Node16",
```
Comment (thombohlk) [inline tsconfig.json:5]: :question: Why was this necessary?

File: src/utils/index.ts (hunk 1 above)
Comment (MPvHarmelen) [reply src/utils/index.ts:13]:
```suggestion
 * Attempt to set a sticky session cookie in the given headers object. If this
```

File: src/Dataset.ts (hunk 2 above)
Comment (MPvHarmelen) [reply src/Dataset.ts:714]: These logs are only printed when someone sets the `DEBUG` environment variable, which I don't think is ever set in (production) ETLs (I'm assuming you mean ETL pipelines).

File: tsconfig.json (hunk 5 above)
Comment (MPvHarmelen) [reply tsconfig.json:5]: VS-code told me and Laurens it was :shrug:

File: src/__tests__/Asset-test.ts (hunk 4 above)
Comment (MPvHarmelen) [reply src/__tests__/Asset-test.ts:48]:
```suggestion
  it.skip("Adding a large asset", async function () {
```

File: src/Dataset.ts (hunk 2 above)
Comment (MPvHarmelen) [reply src/Dataset.ts:714]: See [this line](https://github.com/TriplyDB/TriplyDB-JS/pull/28/files/0340773370fcf0044b83e6e883a61df01340164e#diff-2126a2a0d3a8e8200f8d830a3342c4aadaca2a50f227e5f6a6f1c595cabd865eR7)
